start.py

```diff
@@ -1195,8 +1169,8 @@ def run(self):
         logger.debug(
             "Starting metrics emitter with interval %d", self.interval
         )
-        for line in itertools.cycle(HEARTBEAT_STRING_LIST):
-            logger.info("MENDIX-LOGGING-HEARTBEAT: %s", line)
+        while True:
+            logger.info("MENDIX-LOGGING-HEARTBEAT: .")
```

Comment:
I think the rationale here was to be able to distinguish lines, hence the choice of the Zen of Python (a stupid choice). I would suggest adding an incrementing counter here.

Reply:
Something like this should do it: https://github.com/mendix/cf-mendix-buildpack/pull/204

Reply:
Wasn't the idea behind the counter to have a recognisable value in the logs to make a calculation/estimation on the number of lines missed in the final files. I think we should define what we are actually going to do with this heartbeat information or should keep it simple as just a heartbeat.

Reply:
The original rationale, as I recall it, was to have a well-known message that repeated, so that the output of `cat logs | grep MENDIX-LOGGING-HEATBEAT` would illustrate if a message had been emitted every N hours, and then you could use this (vague) information to diagnose issues in the container (and e.g. figure out if the container had just been started or if it had been running for a long time, based purely on the logs, not `cf events`).

Anyway, the string-based approach, rotating through a list of N strings, is clearly sub-optimal. A counter would at least add some clarity to situations when e.g. hearbeats 3 and 5 are present, but 4 is missing. You could also figure this out based on extrapolating from timestamps, but its harder. Explicit is better than implicit, etc. (Ironically from the Zen of Python)

Reply:
Or remove it all. If the app is not logging it doesn't log. Recent logs can be viewed in live-logging (later) or using "cf logs --recent". It's not that we need the line.

Reply:
I disagree, the value of this is that it conclusively proves that, for a given (coarse) time period, the app was indeed running and logging.

The output `cf logs --recent` doesn't give any guarantees regarding retention. Live logging is only for now onwards. The use case here is proving that the application is indeed running, and showing whether or not the logging infra is running. I have found the presence of the logging heartbeat useful on numerous occasions, since you cannot get this information elsewhere.

Reply:
What about logging a timestamp?